spock-core/src/main/java/org/spockframework/mock/runtime/ProxyBasedMockFactory.java

```diff
@@ -102,11 +109,19 @@ static Object createMock(final Class<?> type,
         new Callable<Class<?>>() {
           @Override
           public Class<?> call() throws Exception {
+            String typeName = type.getName();
+            Class<?> targetClass = type;
+            if (shouldLoadIntoCodegenPackage(type)) {
+              typeName = CODEGEN_PACKAGE + "." + type.getSimpleName();
```

Comment:
should't the type name be the prefix + FQCN, otherwise we might have conflicts

Reply:
AFAIK the type needs to have the same package as the `targetClass`.

This is how Mockito does it:

https://github.com/mockito/mockito/blob/4f72147c464c1a8a642d01fc3334e98e92b464cd/src/main/java/org/mockito/internal/creation/bytebuddy/SubclassBytecodeGenerator.java#L138-L146

https://github.com/mockito/mockito/blob/4f72147c464c1a8a642d01fc3334e98e92b464cd/src/main/java/org/mockito/internal/creation/bytebuddy/SubclassInjectionLoader.java#L42

Reply:
Hm, ok confilcts are avoided via the random Int at the end.

Reply:
The exact package is important here as this is how method handle lookups work. There are no package hierarchies with sub packages, they are just distinct. This is correct.